llvm/lib/TableGen/TGParser.cpp

```diff
@@ -498,6 +503,18 @@ bool TGParser::resolve(const std::vector<RecordsEntry> &Source,
       else
         CheckAssert(E.Assertion->Loc, Condition, Message);
 
+    } else if (E.Dump) {
+      MapResolver R;
```

Comment:
This code is duplicated with above.

Reply:
This code is needed to be able to dump the template parameters of a multiclass instance. In fact, if I remove it, tablegen crashes on the `dump.td` test at the function `llvm::TGParser::resolve`, when resolving the template argument `string s` of the multiclass `MC`.

Reply:
Hum - how could I have put the code for E.dump in the code for E.Assertion? They use different methods...

Anyway - if you are OK in keeping it as it is, please resolve the conversation :)